ray-curator/ray_curator/stages/base.py

```diff
@@ -283,6 +283,26 @@ def ray_stage_spec(self) -> dict[str, Any]:
         """
         return {}
 
+    # --- Custom per-stage metrics helpers ---
+    def _record_metrics(self, mapping: dict[str, float]) -> None:
```

Comment:
I was wondering if we could brainstorm some clearer names for this function. Maybe `set_metrics`, `add_stage_perf`, `add_stage_metrics`, something along those lines? Something that makes it clearer that calling `self._record_metrics({"func1" : time_taken_func1, "func2" : time_taken_func2})` modifies `stage_perf`.